src/lib/viewers/controls/media/VolumeControls.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import classNames from 'classnames';
+import IconVolumeHigh24 from '../icons/IconVolumeHigh24';
+import IconVolumeLow24 from '../icons/IconVolumeLow24';
+import IconVolumeMedium24 from '../icons/IconVolumeMedium24';
+import IconVolumeMute24 from '../icons/IconVolumeMute24';
+import MediaToggle from './MediaToggle';
+import SliderControl from '../slider/SliderControl';
+import useAttention from '../hooks/useAttention';
+import { decodeKeydown } from '../../../util';
+import './VolumeControls.scss';
+
+export type Props = {
+    initialVolume?: number;
+    onVolumeChange: (volume: number) => void;
+    useHotkeys?: boolean;
+};
+
+export const MAX_VOLUME = 1.0;
+export const MID_VOLUME = 0.5;
+export const MIN_VOLUME = 0.0;
+export const STEP_VOLUME = 0.05;
+
+export function getIcon(volume: number): (props: React.SVGProps<SVGSVGElement>) => JSX.Element {
+    let icon = IconVolumeMute24;
+
+    if (volume >= 0.66) {
+        icon = IconVolumeHigh24;
+    } else if (volume >= 0.33) {
+        icon = IconVolumeMedium24;
+    } else if (volume >= 0.01) {
+        icon = IconVolumeLow24;
+    }
+
+    return icon;
+}
+
+export default function VolumeControls({ initialVolume = MAX_VOLUME, onVolumeChange, useHotkeys }: Props): JSX.Element {
+    const [isActive, handlers] = useAttention();
+    const [isMuted, setMuted] = React.useState(!initialVolume);
```

Comment:
Is `isMuted` state needed or can it be derived from `volume`?

Reply:
Not without storing some kind of "previous volume" state. This way seemed simpler.